### ISSUE TITLE
Potential DoS when master key is changed

### ISSUE THREAD

**Issue by githubcdr** [saltstack/salt#39394]
### Description of Issue/Question

Windows minions keep trying to connect when the master key has changed/renewed, when the minion has an outdated minion_master.pub file.

At least 10 connects per second per minion, after manually removing minion_master.pub things calmed down.

### Setup

Latest Salstack

### Steps to Reproduce Issue

Let the Saltmaster create a new key, restart salt-master, accept minions again. Windows minions will DoS salt-master with auth/accept but will never connect.

### Versions Report

latest

**Comment by Ch3LL** [comment]
@githubcdr looks like i'm able to replicate this on 2016.11.2. I was not able to repliccate this on 2016.3.4 on windows. It just stopped the minion processes.

On 2016.11.2 it was attempting to re-auth very frequently, in milliseconds. For example here are the salt/auth timestamps in the master log:

`[DEBUG   ] Sending event: tag = salt/auth; data = {'id': 'win2012', '_stamp': '2017-02-14T17:02:17.454329', 'result': True,`
`[DEBUG   ] Sending event: tag = salt/auth; data = {'id': 'win2012', '_stamp': '2017-02-14T17:02:17.492172', 'result': True,`

And in the minion log it shows:

`[DEBUG ] Connecting to master. Attempt 1840 (infinite attempts) `


Seems its hitting [this](https://github.com/saltstack/salt/blob/develop/salt/minion.py#L550) line, which was added in commit: df3aa0c0 and this pr; https://github.com/saltstack/salt/pull/31794

On a linux VM its not attempting to re-auth as frequently, but does restart the minion process:

```
[CRITICAL] The Salt Master server's public key did not authenticate!
The master may need to be updated if it is a version of Salt lower than 2016.11.2, or
If you are confident that you are connecting to a valid Salt Master, then remove the master public key and restart the Salt Minion.
The master public key can be found at:
/etc/salt/pki/minion/minion_master.pub
[ERROR   ] Error while bringing up minion for multi-master. Is master at localhost responding?
```

In previous releases if the masters key did not authenticate the minion died like so:

```
[CRITICAL] The Salt Master server's public key did not authenticate!
The master may need to be updated if it is a version of Salt lower than 2016.3.4, or
If you are confident that you are connecting to a valid Salt Master, then remove the master public key and restart the Salt Minion.
The master public key can be found at:
/etc/salt/pki/minion/minion_master.pub
[WARNING ] Stopping the Salt Minion
[ERROR   ] Invalid master key
[INFO    ] The salt minion is shutting down..
[INFO    ] The Salt Minion is shut down
The salt minion is shutdown.Invalid master key
 ch3ll@thecakeisalie  ~/git/salt  ➦ 7d79ea7 <B>  
```

 After reading the intention of that PR it seems they wanted to include an option that would re-auth, so this changes the behavior if the masters key is incorrect. One of our engineers will need to look at this to determine what we want the expected default behavior to be when the master key doesn't match (to kill the minion process or enable this new setting by default). But for windows the default should not be re-authing that often in milliseconds. So we definitely need to change that regardless of the architectural decision.

**Comment by Ch3LL** [comment]
@githubcdr would you mind giving the fix in #39424 a try?

**Comment by githubcdr** [comment]
Hi @Ch3LL, I can confirm that this has been fixed in #39424 thanks!
